src/HtmlUtils.js

```diff
@@ -102,10 +102,10 @@ var sanitizeHtmlParams = {
     // URL schemes we permit
     allowedSchemes: ['http', 'https', 'ftp', 'mailto'],
 
-    // DO NOT USE. sanitize-html allows all URL starting with '//'
-    // so this will always allow links to whatever scheme the
-    // host page is served over.
-    allowedSchemesByTag: {},
+    allowedSchemesByTag: {
+        img: [ 'data', 'mxc' ],
```

Comment:
so, i hadn't thought about billion lol attacks here.

perhaps as a compromise, we don't allow data schemas again, but instead we /do/ allow inlined images which refer to MXC URLs (thus making it the media repo's decision as to what's safe to serve to clients).

As you say, allowing it here doesn't help that much, but if the sanitizer goes and fixes up the MXCs into real URLs before inserting them into the page, wouldn't that be okay?

@aviraldg i'm not sure how `data-src` would help us here?